src/cinnamon-app-system.c

```diff
@@ -897,6 +1013,26 @@ cinnamon_app_system_lookup_startup_wmclass (CinnamonAppSystem *system,
   return g_hash_table_lookup (system->priv->startup_wm_class_to_app, wmclass);
 }
 
+/**
+ * cinnamon_app_system_lookup_startup_wmclass_for_flatpak_app:
+ * @system: a #CinnamonAppSystem
+ * @wmclass: (nullable): A WM_CLASS value
+ *
+ * Find a valid flatpak application whose .desktop file contains a
+ * StartupWMClass entry matching @wmclass.
+ *
+ * Returns: (transfer none): A #CinnamonApp for @wmclass
```

Comment:
This should be annotated (nullable) as g_hash_table_lookup can return NULL, as well as the explicit return NULL....unless these are considered error conditions